modules/tinymce/src/core/main/ts/api/UndoManager.ts

```diff
@@ -50,6 +51,27 @@ const UndoManager = (editor: Editor): UndoManager => {
       return Rtc.addUndoLevel(editor, undoManager, index, locks, beforeBookmark, level, event);
     },
 
+    /**
+     * Dispatch a change event if the content in last layer of the undoManager and the current editor content are different
+     *
+     * @method dispatchIfChanged
+     */
+    dispatchIfChanged: (): void => {
+      const data = undoManager.data;
+      const currentLevel = Levels.createFromEditor(editor);
+
+      Arr.last(data).filter((lastLevel) => {
```

Comment:
This doesn't account for the `index` property which says which undo level we should be looking at currently (which can move backwards or forwards as you undo/redo changes). This probably needs to use similar logic to `addUndoLevel` from `Operations.ts`, e.g:
```
const lastLevel = undoManager.data[index.get()];
```

Another issue I see here now looking at this, is that it won't work if the `lastLevel` is `undefined`. It probably still should be firing in that case, so we should add a test for that as well.

Reply:
That was well spotted.

- [ ] Add a test that does a change then undo then run this then it should not fire a change event
- [ ] Add a test that clears all undo levels then it should always fire I guess since we have nothing to compare against

Reply:
done, instead of `data[index.get()]`, I used `Arr.get` I hope is ok